src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -396,6 +408,22 @@ private bool filterCallback(Func<Document, DocumentFlags, bool> filterFunction,
              return filterFunction(new Document(Config.Database, docID, value), flags);
         }
 
+        private bool PermanentNetworkError(C4Error error)
+        {
+            // If this is a transient error, or if I'm continuous and the error might go away with a change
+            // in network (i.e. network down, hostname unknown), then go offline and retry later
+            var transient = Native.c4error_mayBeTransient(error) ||
+                            (error.domain == C4ErrorDomain.WebSocketDomain && error.code ==
+                             (int)C4WebSocketCustomCloseCode.WebSocketCloseUserTransient);
+
+            if (!transient && !(Config.Continuous && Native.c4error_mayBeNetworkDependent(error))) {
+                WriteLog.To.Sync.I(Tag, "Permanent error encountered ({0} / {1}), giving up...", error.domain, error.code);
+                return true; // Nope, this is permanent
+            }
+
+            return false;
+        }
+
```

Comment:
I suggest reusing this in the `HandleError` method, and move the logging outside of this method so it doesn't happen twice.